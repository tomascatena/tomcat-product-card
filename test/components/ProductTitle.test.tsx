import React from 'react';
import { ProductTitle } from '../../src/components/ProductTitle';
import renderer from 'react-test-renderer';
import { PRODUCT_ONE } from '../data/products';
import { ProductCard } from '../../src/components/ProductCard';

describe('ProductTitle', () => {
  it('should render the component with the custom title', () => {
    const wrapper = renderer.create(<ProductTitle title="Custom Product" />);

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should render the component with the name if the product', () => {
    const wrapper = renderer.create(
      <ProductCard product={PRODUCT_ONE}>
        {
          () => (
            <>
              <ProductTitle />
            </>
          )
        }
      </ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});