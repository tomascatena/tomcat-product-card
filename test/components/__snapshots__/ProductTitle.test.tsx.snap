// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle should render the component with the custom title 1`] = `
<span
  className="productDescription undefined"
>
  Custom Product
</span>
`;

exports[`ProductTitle should render the component with the name if the product 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - Without Image
  </span>
</div>
`;
